hudi-utilities/src/test/java/org/apache/hudi/utilities/deltastreamer/TestTransformer.java

```diff
@@ -79,6 +96,71 @@ public void testMultipleTransformersWithIdentifiers() throws Exception {
     assertEquals(0, sqlContext.read().format("org.apache.hudi").load(tableBasePath).where("timestamp != 110").count());
   }
 
+  @Test
+  public void testTransformerSchemaValidation() throws Exception {
+    List<String> transformerClassNames = Arrays.asList(
+        FlatteningTransformerWithTransformedSchema.class.getName(),
+        TimestampTransformer.class.getName(),
+        AddColumnTransformerWithTransformedSchema.class.getName());
+    runDeltaStreamerWithTransformers(transformerClassNames);
+  }
+
+  @Test
+  public void testTransformerSchemaValidationFails() throws Exception {
+    String errorMsg = "Expected target schema not provided for transformer";
+    List<String> transformerClassNames = Arrays.asList(
+        FlatteningTransformer.class.getName(),
+        TimestampTransformer.class.getName(),
+        AddColumnTransformerWithTransformedSchema.class.getName());
+    testTransformerSchemaValidationFails(transformerClassNames, errorMsg);
+
+    transformerClassNames = Arrays.asList(
+        FlatteningTransformerWithTransformedSchema.class.getName(),
+        TimestampTransformer.class.getName(),
+        AddColumnTransformer.class.getName());
+    testTransformerSchemaValidationFails(transformerClassNames, errorMsg);
+
+    errorMsg = "Schema of transformed data does not match expected schema";
+    transformerClassNames = Arrays.asList(
+        FlatteningTransformerWithTransformedSchema.class.getName(),
+        TimestampTransformer.class.getName(),
+        AddColumnTransformerWithWrongTransformedSchema.class.getName());
+    testTransformerSchemaValidationFails(transformerClassNames, errorMsg);
+  }
+
+  private void testTransformerSchemaValidationFails(List<String> transformerClasses, String errorMsg) {
+    try {
```

Comment:
You can use assertThrows from junit for thishttps://junit.org/junit5/docs/5.8.2/api/org.junit.jupiter.api/org/junit/jupiter/api/Assertions.html#assertThrows(java.lang.Class,org.junit.jupiter.api.function.Executable,java.lang.String)